### PR TITLE
Implement UpdateUsers function

### PR DESCRIPTION
**Pull request by justinmuskopf**

### CODE REVIEW

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 1]
```diff
@@ -87,9 +87,30 @@ function cfcuser:UserExists(user)
...
 
-function cfcuser:UpdateUser(user, lastonline, factionid, kills, deaths, factionrank, hoursinfaction)
+local function IsValidNumber(num)
+    return IsValidAndOfType(num, 'number')
+end
 
+local function IsValidString(str)
+    return IsValidAndOfType(str, 'string')
+end
+
```
Comment (justinmuskopf) [inline lua/cfcfactions/core/sv/sv_users.lua:101]: These `IsValid` something-or-rathers could go in `sh_sharedutils.lua` if they'll be useful elsewhere... 
There's also **CFC LIB**

**Review by ghost — COMMENTED** [review]
Looks good

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 2]
```diff
@@ -87,9 +87,30 @@ function cfcuser:UserExists(user)
 	end
 end
 
+local function IsValidAndOfType(item, t)
+    return item ~= nil and type(item) == t
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua:91]: If we never use this for booleans we can just `return item and type(item) == t`

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 3]
```diff
@@ -87,9 +87,30 @@ function cfcuser:UserExists(user)
...
-function cfcuser:UpdateUser(user, lastonline, factionid, kills, deaths, factionrank, hoursinfaction)
+local function IsValidNumber(num)
+    return IsValidAndOfType(num, 'number')
+end
 
+local function IsValidString(str)
+    return IsValidAndOfType(str, 'string')
+end
+
+function cfcuser:UpdateUser(user, lastonline, factionid, kills, deaths, factionrank, hoursinfaction)
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua:102]: Can we go ahead and properly camel these?

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 4]
```diff
@@ -87,9 +87,30 @@ function cfcuser:UserExists(user)
...
+
+    local userTable = cfcuser[user:SteamID64()]
+    if IsValidString( lastonline ) then userTable["LastOnline"] = lastonline end
+    if IsValidNumber( factionId ) and cfcFactions:IsValidFaction( factionid ) then 
+        userTable["FactionId"] = factionid 
+    end    
+    if IsValidNumber( kills ) then userTable["Kills"] = kills end
+    if IsValidNumber( deaths ) then userTable["Deaths"] = deaths end
+    if IsValidString( factionrank ) then userTable["FactionRank"] = factionrank end
+    if IsValidNumber( hoursinfaction ) and hoursinfaction > 0 then userTable["HoursInFaction"] = hoursinfaction end
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua:113]: References mess me up, but will these be changing the actual `cfcuser` table, or will it be changing a copied version if it in the `userTable` var?

**Review by sarahsturgeon — APPROVED** [review]
Approving, just a few discussion topics.


Also, will we need to do this elsewhere? This seems like a process that could/should be extracted into its own DataValidation tool.

File: lua/cfcfactions/core/sv/sv_users.lua (hunk 4 above)
Comment (ghost) [reply lua/cfcfactions/core/sv/sv_users.lua:113]: Actual changing. Its intended to update a player on the server. Their inner details of a faction

**Pull request merged**